homeassistant/components/environment_canada/sensor.py

```diff
@@ -28,42 +28,67 @@
 
 CONF_ATTRIBUTION = "Data provided by Environment Canada"
 CONF_STATION = 'station'
+CONF_LANGUAGE = 'language'
 
 MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(minutes=10)
 
 SENSOR_TYPES = {
-    'temperature': {'name': 'Temperature',
+    'temperature': {'english': 'Temperature',
```

Comment:
It feels counterproductive to me to put translations, hardcoded, in the backend.
A user can customize entities if he wants to, right?

Reply:
Yes, they can. In this case the source data is available in both [English](http://dd.weatheroffice.ec.gc.ca/citypage_weather/xml/ON/s0000430_e.xml) and [French](http://dd.weatheroffice.ec.gc.ca/citypage_weather/xml/ON/s0000430_f.xml), and I'd like a user to be able to select one of the two languages and have both the labels and the underlying data match.

Is there a way to leverage the user selection for the front end language for this?

Reply:
We shouldn't merge this, as translations are not stored inside the code.

Reply:
Thanks for the feedback. I've moved the labels out to the `env_canada` library, which makes things cleaner. Please let me know what you think.

Reply:
That is just working around the original concern raised!?

Reply:
I think the original concern is valid, so I've tried to address [the issue](https://github.com/michaeldavie/env_canada/issues/2) in a more logical way. The original data source supports two languages, and now the library for interacting with it does as well. 

This approach minimizes the impact on Home Assistant:
* There's a change to how the data returned is structured.
* An optional language parameter is added to the sensor platform.
* The names and units of sensor entities are no longer hardcoded in the backend, and are instead based on the data returned.

Reply:
We should not have the names change based on the language. That's not how translations are managed within Home Assistant, nor are we able to create an exception here. Exceptions lead to more messiness down the road. I suggest instead that you implement a unique ID so that entities can be renamed by users, including into languages beyond French and English.

Reply:
Thanks, I believe I've addressed this in the commits below. In the latest version the `entity_id` values are fixed, and are based on the keys returned by the external library. The user can also specify a `platform_name` to prepend to these as group.

https://github.com/home-assistant/home-assistant/blob/b92bed9cc9fa9ffd60e35945054bdef72cb708da/homeassistant/components/environment_canada/sensor.py#L92-L95

The `friendly_name` values that are displayed in the GUI are also returned by the external library, based on whether the platform has been configured for English or French. Users can change these to whatever they like.

https://github.com/home-assistant/home-assistant/blob/b92bed9cc9fa9ffd60e35945054bdef72cb708da/homeassistant/components/environment_canada/sensor.py#L128

Please take a look at the latest version and let me know if this works.

Reply:
unique ID != entity_id

Reply:
Sorry, but I'm afraid I don't understand. Which property would you like me to change? Is there another sensor that has implemented this the correct way that you can point me to?

Reply:
`unique_id` is a property on the entity that can be implemented to activate the entity registry for the entity.

https://developers.home-assistant.io/docs/en/entity_index.html#generic-properties

https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

https://developers.home-assistant.io/docs/en/entity_registry_index.html

Reply:
Sorry, I should have given more context. Thanks Martin.

Reply:
Thanks for clarifying, I've switched this over.